browser/js/common/directives/graphs/add-graph.directive.js

```diff
@@ -0,0 +1,77 @@
+app.directive('addGraph', function($http) {
+    return {
+        restrict: 'E',
+        templateUrl: 'js/common/directives/graphs/add-graph.directive.html',
+        scope: {
+            data: "=",
+            info: "=",
+            columns: "="
+        },
+        link: function(scope, ele, attrs) {
+
+        	function validateNumber(col) {
+        		let numArray = [];
+        		for (let i=0; i < scope.data.length; i++) {
+        			if (i === 10) break;
+        			else {
+        				if (!isNaN(scope.data[i][col.name])) numArray.push(scope.data[i]);
```

Comment:
It looks like you're only pushing to the array in order to check whether it has length later. So why not just `return true` here instead of pushing to the array, and after the end of the loop putting `return false`.